---
description: Written by students for students.
---

# Preface

## Brief Introduction

Welcome to my workbook! This is the brief part where I can tell you a bit about my own experience in genomics and bioinformatics.

I was hired at the Applied Genomics Centre in second-year as a part-time lab assistant for the summer. I went through the same introductory wet-lab on-boarding of PCR and sequencing (with some FLA mixed in too). As I racked up sequencing runs on our humble SeqStudio (and racked lots of tips), my summer project eventually reached the data analysis stage. I stumbled into the world of `Python`, `R`, and processing lots and lots of `.csv` files. I quickly saw that this field of data analysis was an unfilled niche—and something I could get really good at.

Since then, I've been involved in multiple bioinformatics projects at the AGC, especially those involved in bacteria. I've done microbial composition analysis, large-scale variant-calling, protein-folding, genome assembly, and more—alongside many pure programming projects.&#x20;

Altogether, I am writing this workbook to provide other learners with some direction in learning about bioinformatics.

## Target audience

This workbook was written for readers to get a brief introduction into bioinformatics as it is applied at the AGC (with a focus on agricultural genomics). Whether you are coming in as a new bioinformatics student or just focusing on the essentials, this workbook is intended to provide a comprehensive and accessible resource for grasping the fundamentals and developing essential skills.

Again, repeating that this work is "applied", the content described herein are things I have personally used in some project at the AGC.

## Recommended hardware

Bioinformatics (almost by definition) involves working with large datasets and performing some very computationally intensive tasks. That said, this introductory workbook mostly involves online tools and lightweight software. Almost any computer from this decade should be sufficient to work through the book.

As for your operating system (OS), most users in the lab have a work-provided laptop running Windows. Some have macOS. Even fewer are running some distribution of Linux. Most of the content described here has been extensively tested on Windows, though I would recommend you get familiar with a Linux-based system for the more advanced content.

## Scope and Approach

I've designed this workbook to provide a comprehensive learning experience, but you don't need to complete every exercise to benefit from it. Rather, the book serves as a reference for honing specific skills and techniques relevant to specific projects. I'm hoping that with the sections and exercises, you can easily pinpoint the areas that best support your learning and career goals.

## Prerequisites

None! While no prior knowledge of bioinformatics is required, it would be useful to have some understanding of foundational molecular biology principles: DNA, RNA, transcription, translation. At the least, it would be good to know the basic alphabet: `A`, `T`, `C`, `G`, (and sometimes `U`).

So to give some specifics, it would be good to have BIOL2320 (2nd year genetics), but I will try to include sections for biological context where necessary.

## Acknowledgements

I'd like to thank Spruce's 2nd-floor vending machine for supplying me with the caffeinated drinks to get through my undergrad.
