# Table of contents

* [Home](README.md)
* [Preface](preface.md)

## I. Essentials <a href="#i-essentials" id="i-essentials"></a>

* [1: Introduction to Bioinformatics](i-essentials/1-introduction-to-bioinformatics.md)
* [2: The NCBI Toolset](i-essentials/2-the-ncbi-toolset/README.md)
  * [2.1: Retrieving sequences from NCBI](i-essentials/2-the-ncbi-toolset/2.1-retrieving-sequences-from-ncbi.md)
  * [2.2: BLAST](i-essentials/2-the-ncbi-toolset/2.2-blast.md)
  * [2.3: Primer-BLAST](i-essentials/2-the-ncbi-toolset/2.3-primer-blast.md)
* [3: Sanger Sequencing & ApE (AplasmidEditor)](i-essentials/3-sanger-sequencing-and-ape-aplasmideditor/README.md)
  * [3.1: AplasmidEditor (ApE) - Introduction and installation](i-essentials/3-sanger-sequencing-and-ape-aplasmideditor/3.1-aplasmideditor-ape-introduction-and-installation.md)
  * [3.2: Exploring an electropherogram](i-essentials/3-sanger-sequencing-and-ape-aplasmideditor/3.2-exploring-an-electropherogram.md)
  * [3.3: Pairwise alignments](i-essentials/3-sanger-sequencing-and-ape-aplasmideditor/3.3-pairwise-alignments.md)
  * [3.4: Annotations and the GenBank format](i-essentials/3-sanger-sequencing-and-ape-aplasmideditor/3.4-annotations-and-the-genbank-format.md)
  * [3.5: ORFs & translation](i-essentials/3-sanger-sequencing-and-ape-aplasmideditor/3.5-orfs-and-translation.md)
  * [3.6: ApE extras](i-essentials/3-sanger-sequencing-and-ape-aplasmideditor/3.6-ape-extras.md)
* [4: Multiple-sequence alignment (MSA) & JalView](i-essentials/4-multiple-sequence-alignment-msa-and-jalview/README.md)
  * [4.1: JalView - Introduction and installation](i-essentials/4-multiple-sequence-alignment-msa-and-jalview/4.1-jalview-introduction-and-installation.md)
  * [4.2: Multiple sequence alignments (MSA)](i-essentials/4-multiple-sequence-alignment-msa-and-jalview/4.2-multiple-sequence-alignments-msa.md)
  * [4.3: Phylogeny](i-essentials/4-multiple-sequence-alignment-msa-and-jalview/4.3-phylogeny.md)
* [5: Online Genome Browsers](i-essentials/5-online-genome-browsers/README.md)
  * [5.1: NCBI](i-essentials/5-online-genome-browsers/5.1-ncbi.md)
  * [5.2: Ensembl](i-essentials/5-online-genome-browsers/5.2-ensembl.md)
  * [5.3: UCSC](i-essentials/5-online-genome-browsers/5.3-ucsc.md)
* [6: Next-Generation Sequencing (NGS)](i-essentials/6-next-generation-sequencing-ngs.md)

## II. Linux and the Command-line <a href="#ii-linux-and-cli" id="ii-linux-and-cli"></a>

* [Recommended resources](ii-linux-and-cli/recommended-resources.md)

## III. Python and programming <a href="#iii-python-and-programming" id="iii-python-and-programming"></a>

* [Recommended resources](iii-python-and-programming/recommended-resources.md)

## Appendices

* [Appendix A: Biological Background](appendices/appendix-a-biological-background/README.md)
  * [Review of Sanger Sequencing](appendices/appendix-a-biological-background/review-of-sanger-sequencing.md)
* [Appendix B: Exercises](appendices/appendix-b-exercises/README.md)
  * [1.2.2A: Fish identification (BIOL 3320)](appendices/appendix-b-exercises/1.2.2a-fish-identification-biol-3320.md)
  * [1.2.2B: Bacterial identification](appendices/appendix-b-exercises/1.2.2b-bacterial-identification.md)
* [Appendix C: Projects](appendices/appendix-c-projects.md)
