# 4: Multiple-sequence alignment (MSA) & JalView

{% hint style="info" %}
## Learning Objectives

### 4.1: JalView - Introduction and installation

* [ ] Understand the importance of sequence alignment in bioinformatics research and its applications in comparative genomics.
* [ ] Describe the function and purpose of JalView software.
* [ ] Use JalView to perform basic sequence analysis tasks, such as calculating sequence statistics and filtering sequence data.&#x20;



### 4.2: Multiple sequence alignments (MSA)

* [ ] Perform sequence alignment with Jalview using various alignment algorithms, and identify the strengths and limitations of each approach.
  * [ ] Muscle
  * [ ] Clustal
  * [ ] MAFFT
* [ ] Compare multiple sequence alignments with pairwise alignments and identify their purposes.
* [ ] Identify conserved regions conserved regions of the alignment and recognize their significance in sequence conservation.



### 4.3: Phylogeny

* [ ] Understand the fundamental concepts and terminologies used in phylogenetic analysis, such as branch lengths, evolutionary distance, and tree topology.
* [ ] Generate phylogenetic trees from the sequence alignment, using various methods such as neighbor-joining and average distance.
* [ ] Evaluate reliability of the phylogenetic tree using bootstrapping, posterior probabilities, or other methods.
* [ ] Interpret the phylogenetic tree topology and the branch lengths, identify the relationships among the sequences, and recognize the significance of any patterns or clusters observed in the tree.
{% endhint %}
