---
description: >-
  Find your match with NCBI BLAST: Identifying similar sequences faster than a
  bioinformatician can say 'homology'!
---

# 2.2: BLAST

{% hint style="info" %}
**Learning Objectives**

* [ ] Understand the basic principles behind alignments and sequence similarity search.&#x20;
* [ ] Perform a BLAST with the NCBI nucleotide database.
* [ ] Interpret the quality and reliability of BLAST results.
* [ ] Explore BLAST programs and their use-cases.
  * [ ] blastn
  * [ ] blastp
  * [ ] tblastn, blastx, tblastx
{% endhint %}

## Background

NCBI [BLAST ](#user-content-fn-1)[^1]\(Basic Local Alignment Search Tool) is a widely-used bioinformatics software tool that allows for the rapid comparison of nucleotide or protein sequences against a vast array of publicly available sequence databases. BLAST provides a flexible and powerful method for identifying sequence similarities and differences, aiding in the discovery of new genes, functional domains, and evolutionary relationships.\
\
There are different kinds of BLAST services that are available on NCBI:

* **blastn:** Search a nucleotide sequence against a database of nucleotide sequences.
* **blastp:** Search a protein sequence against a database of protein sequences.
* **tblastn, blastx, tblastx**: Search with a nucleotide or protein sequence and against databases using translation or reverse-translation.

{% hint style="info" %}
Can you think of some examples where one BLAST program might be better over another? Or situations where you would use a specific BLAST program?
{% endhint %}

### Definitions, Part I

As with anything in Biology, there are definitions for specific items.

* **Query sequence**: The sequence for which you are looking for similarities. The sequence that is being checked against the database.
* [**Subject sequence**](#user-content-fn-2)[^2]**:** A sequence that exists in the database that has some similarity to your query sequence.
* **Pairwise alignment:** This is an alignment where 2 sequences are compared. Generally, the [query sequence](#user-content-fn-3)[^3] is compared against a subject sequence. Alignments generally try to match up as much of the sequence as possible.

### The algorithm

The way that BLAST works is that it aligns the query sequence against subject sequences in a database. Each of the letters in the query sequence (whether it be nucleotide or amino acid) are paired to the subject sequence so that there is either a perfect _match, mismatch,_ or _gap._\
\
Consider the following sequences:

```
Sequence 1: AATCTATA
Sequence 2: AAGATA
```

{% hint style="info" %}
How can these sequences be aligned to show regions of similarity?
{% endhint %}

The annoying answer is: it depends. There are actually a lot of ways to align those sequences and have some sort of similarity shown. Consider that each of the alignments below are possible configurations.

```
AATCTATA    AATCTATA    AATCTATA
||##||       |###|#       ###|||
AAGATA       AAGATA       AAGATA
```

Take a look at each of the tabs below to find out more information about this alignment.

{% hint style="info" %}
Look at each of the alignments. Compare them in terms of the _matches, mismatches,_ and _gaps._ Which one would you say is the best alignment?
{% endhint %}

{% tabs %}
{% tab title="Alignment #1 (Left)" %}
There are `4` matches, `2` mismatches, and `2` unaligned nucleotides.&#x20;
{% endtab %}

{% tab title="Alignment #2 (Middle)" %}
There are `2` matches, `4` mismatches, and `2` unaligned nucleotides.
{% endtab %}

{% tab title="Alignment #3 (Right)" %}
There are `3` matches, `3` mismatches, and `2` unaligned nucleotides.
{% endtab %}
{% endtabs %}

Since they all have the same number of unaligned nucleotides, the one that wins out is the one with most matches and least mismatches: Alignment #1. This sequence also happens to pass the eye test — it looks like it would be the best alignment and BLAST would agree.\
\
But wait, we've just shifting the `AAGATA` around. What about gaps?

```
AATCTATA    AATCTATA    AATCTATA
||# ## |    || # |||    ||  #|||
AAG-AT-A    AA-G-ATA    AA--GATA
```

Now considering the gaps, the idea of the _best_ alignment starts to get fuzzy.

{% hint style="info" %}
Consider these alignments and think about the _matches_, _mismatches_, and _gaps_. Also consider the biological meaning behind a mismatch or gap. Which one of these alignments probably makes the most sense?
{% endhint %}

I'm sure you have a subjective idea about which one is best here. But for bioinformatics, it is more important to have _reproducibility_ and _consistency_. That is to say, the parameters when performing an operation, such as alignment, should not change unpredictably. For this, computers can be used to perform alignments both algorithmically and iteratively with scores assigned to matches, mismatches, and gaps.

{% hint style="warning" %}
**Quick note:** The actual values used in scoring will not be covered in this lecture, but if you are interested, I recommend checking out this review on [BLOSUM62](https://www.nature.com/articles/nbt0804-1035).
{% endhint %}

Using all these scoring metrics, BLAST then identifies which subject sequences in the database are closest to your query sequence.

## Performing a nucleotide BLAST

{% hint style="info" %}
NCBI's web BLAST tools can be accessed with this link: [https://blast.ncbi.nlm.nih.gov/Blast.cgi](https://blast.ncbi.nlm.nih.gov/Blast.cgi)
{% endhint %}

After navigating to the link, select Nucleotide BLAST to search a query nucleotide sequence against databases of nucleotides sequences.

<div>

<figure><img src="https://files.gitbook.com/v0/b/gitbook-x-prod.appspot.com/o/spaces%2FuXQehO6lArpiroG7tIdI%2Fuploads%2Fuu5aJ2Nydo26gxt8KoYf%2Fimage.png?alt=media&#x26;token=5ad57f03-10ba-4910-a313-332b86c78edf" alt=""><figcaption><p><strong>Figure 1.</strong> The main landing page for NCBI BLAST showing various BLAST programs.</p></figcaption></figure>

 

<figure><img src="https://files.gitbook.com/v0/b/gitbook-x-prod.appspot.com/o/spaces%2FuXQehO6lArpiroG7tIdI%2Fuploads%2Fr3ET5HDt8juD3KzrLsM3%2Fimage.png?alt=media&#x26;token=87952735-e12b-46f0-a8a7-390e1d640cbc" alt=""><figcaption><p><strong>Figure 2.</strong> The main search page for performing NCBI BLASTN to search nucleotide database </p></figcaption></figure>

</div>

The most obvious thing to do first is input [a sequence we want to search against the database](#user-content-fn-4)[^4]. For the purposes of this textbook, I'll give you a sequence here to look up.

{% code overflow="wrap" %}
```
>Mystery rbcL sequence (684 nt)
TTCAAAGCTGGTGTTAAAGATTATAAATTGACTTATTATACTCCTGACTATGAAACCAAAGATACTGATATCTTGGCAGCATTCCGAGTAACTCCTCAACCAGGAGTTCCGCCTGAAGAAGCAGGTGCCGCGGTAGCCGCCGAATCTTCTACTGGTACATGGACAACTGTGTGGACCGATGGGCTTACCAGTCTTGATCGTTACAAAGGGCGATGCTACGGCCTTGAACCTGTTGCTGGGGAAGAAAATCAATATATTGCTTATGTAGCTTATCCCTTAGACCTTTTTGAAGAAGGTTCTGTTACTAACATGTTTACTTCCATTGTCGGTAATGTATTTGGGTTCAAGGCCCTGCGTGCTCTACGTCTGGAGGATTTGCGAATCCCTACTGCTTATATTAAAACTTTCCAAGGTCCGCCTCATGGCATCCAAGTTGAGAGAGATAAATTGAACAAGTATGGTCGTCCCCTATTAGGATGTACTATTAAACCTAAATTGGGGTTATCCGCTAAGAATTATGGTAGAGCTGTTTATGAATGTCTTCGTGGGGGACTTGATTTTACCAAAGATGATGAAAATGTGAATTCCCAACCATTTATGCGTTGGAGAGACCGTTTCTTATTTTGTGCCGAAGCCATTTTTAAATCACAGGCTGAAACAGTGAAATCAAAGGCATTACTTGAT
```
{% endcode %}

Go ahead and put that into the box. Alternatively, we could have also pasted in an accession or GI number if we had it, or uploaded a file in FASTA format.\
\
From here, there a couple parameters we can set for our BLAST search:

* **Query Subrange**\
  Within our query sequence, we can tell the program to look at a specific window within it.
* **Job title**\
  When submitting multiple queries at a time, it can be useful to set a name for the job to discriminate which search results belong to which job.
* **Database**\
  We can select specifically which database to search against. Most commonly, we search against the standard collection of all nucleotide entries (nr/nt).
* **Exclusions**\
  We can also exclude certain organisms from our search results. This is especially relevant for excluding environmental samples which have no taxonomic information. In primers, for example, we may want to exclude our target organism to limit our results to only off-target organisms.
* **Entrez Query**\
  From the previous chapter, we discussed search terms. Conveniently enough, we can also use those search terms here to create a custom database.
* **BLAST algorithm**\
  BLAST offers a choice of algorithm, depending on the query and matches that you are expecting. For most of what we do, `Highly similar sequences (megablast)` is optimal.

{% hint style="info" %}
**Scenario:**

A man named Jack has come to our lab looking for some scientific advice. He has bought some seeds where the salesman claims that each seed is imbued magical properties.&#x20;

We've ground these up and performed a DNA extraction. Seeing as they're eukaryotic plant DNA, we've decided to sequence [_rbcL_](#user-content-fn-5)[^5]_._\
\
Using the `mystery rbcL sequence`, perform a nucleotide BLAST to see if we can identify this strange seed.



**Hints:**&#x20;

* In terms of the database, should we be using a database of just reference sequences?
* Is there a specific range within the sequence we should use for our search or should we just search using the entire sequence?
{% endhint %}

## Interpreting the results

After you've submitted your search, it should take a minute or so for it to search your selected database. Once it is finished, your browser should redirect to a results page that looks like the one below (Figure 3).

<figure><img src="https://files.gitbook.com/v0/b/gitbook-x-prod.appspot.com/o/spaces%2FuXQehO6lArpiroG7tIdI%2Fuploads%2Fp5rZVOs4OvbsmdaehlLv%2Fimage.png?alt=media&#x26;token=bb4a7670-8d5e-48b5-b60f-6b7c0043caeb" alt=""><figcaption><p><strong>Figure 3.</strong> An example results page from NCBI BLASTN.</p></figcaption></figure>

### Definitions, Part II: Electric boogaloo

Keep in mind that BLAST works by comparing your query sequence to each subject sequence in a _pairwise_ fashion, i.e. one-to-one.&#x20;

* **% query coverage**\
  This is the percent of the query sequence that is included in the aligned segments.
* **% identity**\
  This describes how similar the aligned segments of the query are to the given subject sequence.
* **E-value**\
  This is probability that the alignment occurred by chance alone — that it's just by chance that these sequences are similar. If you have a long sequence with a lot of alignment, it becomes slightly tougher to argue that the query and subject just happen to share the same sequence by random chance. In other words, lower E-value means higher confidence of a match.

**The matchmaker's recipe:** To be as confident as possible in a match, you want:

* High % query coverage.
* High % identity.
* E-value as close as possible to 0.

{% hint style="info" %}
**Scenario:**

Now it's your turn to interpret your results from before. We submitted a query for our mystery rbcL sequence. Look at the description and scientific name for the organisms in the entries. \
\
What organisms seem to be our likely match? Can we be confident in those results? Explain why or why not.

\
Some of those values (% query coverage, % identity, E-value) might not be optimal. Can you think of why [they may not be perfect](#user-content-fn-6)[^6]?
{% endhint %}

<details>

<summary>Extra practice</summary>

If you're itching for some extra practice, it seems like Jack isn't totally convinced.\
\
He wants us to perform some additional sequencing to verify the identity we have uncovered. This time, we have decided to sequence the [_ITS_](#user-content-fn-7)[^7] region from these seeds. \
\
Work through the steps for completing a BLAST again and try to verify the identity of Jack's strange seeds.

{% code overflow="wrap" %}
```
>Mystery ITS sequence (375 nt)
CGATGCCTCACAATCAGATTGACCCGCGAACTTGTTTATTCATCTACCGTCGGGAGGGAGGGGATGACCACGGCGCCCCGTGCGCCCGGCCTCCTCGTCCTCGCGACAAACACAAACCCCGGCGCTTCGTGCGCCAAGGAACTCAAATCTGTTAAGTGCGACTCCCGGGGGCCCGGAGACGGTGTCCCGCGGGAGTCGTCACGACACAACATTTACATACAATGACTCTCGGCAACGGATATCTCGGCTCTTGCATCGATGAAGAACGTAGCGAAATGCGATACTTGGTGTGAATTGCAGAATCCCGTGAACCATCGAGTCTTTGAACGCAAGTTGCGCCCGAAGCCATTAGGCCGAGGGCACGCCTGCCTGGGT
```
{% endcode %}

</details>

## blastn, blastp, tblastn, blastx, tblastx&#x20;

We've performed a BLAST search with the most common and straight-forward program: blastn. Like I mentioned earlier, there are several versions of BLAST, each designed for different types of sequences and applications.

* **BLASTN** is used for nucleotide-to-nucleotide searches and is commonly used to compare DNA sequences.
* **BLASTP** is used for protein-to-protein searches and is commonly used to compare amino acid sequences.
* **TBLASTN, BLASTX, TBLASTX** are used when you need to translate between sequences in your search. BLASTX is used to search a nucleotide sequence by translation against databases of proteins. TBLASTN and TBLASTX are used to go the other way: searching a protein against nucleotide databases.

{% hint style="info" %}
If we have the amino acid sequence for a given protein, which of those versions of BLAST would you want to use and why?&#x20;
{% endhint %}

## Summary

NCBI BLAST (Basic Local Alignment Search Tool) is a widely used program for comparing biological sequences. There are several versions of BLAST on NCBI, each designed for different types of sequences and applications.&#x20;

* BLASTN is used for comparing nucleotides to nucleotides.
* BLASTP is used for comparing proteins to proteins.
* TBLASTN, BLASTX, TBLASTX are used when there is translation involved (protein-to-nucleotide and vice-versa).

BLAST results must be interpreted by looking at alignment values: % query coverage, % identity, and E-value. A common use for BLAST is identification of unknown organisms.&#x20;

## Suggested Assignments

{% hint style="success" %}
Try these out for some extra practice with NCBI BLAST:

[1.2.2a-fish-identification-biol-3320.md](../../appendices/appendix-b-exercises/1.2.2a-fish-identification-biol-3320.md "mention")

[1.2.2b-bacterial-identification.md](../../appendices/appendix-b-exercises/1.2.2b-bacterial-identification.md "mention")
{% endhint %}

[^1]: Just like how you can Google something, we often say that you can BLAST something, or that you BLASTed this or that. Funny how words work.

[^2]: Also sometimes referred to as the reference sequence.

[^3]: See above.

[^4]: Again, this is the query sequence.

[^5]: Also known as the large subunit of RuBisCo, this protein is involved in the Calvin cycle for plants. When this protein's mom is angry, she uses its full name: ribulose-1,5-bisphosphate carboxylase/oxygenase large subunit.

[^6]: **Hint:** stupid biological variation. Always ruining things. Or maybe some technique-related reason?

[^7]: The internal trascribed spacer (ITS) is a small region of DNA spaced between the small-subunit and large-subunit ribosomal rRNA genes. This is more commonly used for fungi, but we can still use it here.
