---
description: Adapted from the fish lab, but bacterial.
---

# 1.2.2B: Bacterial identification

## Objective

To identify the species of various samples of bacteria that have been isolated and sub-cultured from a bioreactor.

## Required and recommended software

None! You should be able to complete this with just web-based resources.

## Background

Bacterial identification is a fundamental aspect of microbiology, with implications in various fields, including medicine, biotechnology, and environmental science. The traditional methods of bacterial identification rely on phenotypic characteristics, such as morphology, biochemical reactions, and growth patterns. \
\
As a personal anecdote, I absolutely loved the idea of using the catalase test and the Vogesâ€“Proskauer (VP) test in Microbiology I. But of course, I wondered if there was an easier way to identify bacteria without creating a dichotomous tree and process-of-elimination.\
\
The 16S rRNA gene codes for the 16S ribosomal RNA component of the 30S subunit of the prokaryotic ribosome. If you can imagine the importance of transcription (and being able to make proteins), this gene is present in all species of bacteria. It is conserved enough that we can use universal primers to amplify regions of it, while variable enough that we can use it for identification. One might even say _hypervariable._

<figure><img src="https://files.gitbook.com/v0/b/gitbook-x-prod.appspot.com/o/spaces%2FuXQehO6lArpiroG7tIdI%2Fuploads%2FtPBjsD4pmsxwGbpIH2FO%2Fimage.png?alt=media&#x26;token=ba9234a7-497b-4656-8ca6-924893d454c6" alt=""><figcaption><p><strong>Figure 1.</strong> An overview of the 9 hypervariable regions located in the bacterial 16S rRNA gene. Notice as well the primers that may be used to amplify and/or sequence parts of this gene.</p></figcaption></figure>

For this project, our collaborators have noticed that a few of their bioreactors are clearly outperforming the others. They've isolated some of the bacteria in one of those hyper productive bioreactors and plated it for us to identify. A DNA extraction is performed and the 16S rRNA gene is amplified using the 27F primer. The sequencing results in FASTA format are below.

<details>

<summary>Benchtop Methods</summary>

Someone remind me to do this.

</details>

## Analysis

{% tabs %}
{% tab title="Bacterial isolate A" %}
{% code overflow="wrap" %}
```
>Bacterial isolate A
SDRBBDGATCTGCCGGAGGCCACTGCTATTGGGTTTCGATTAAGCCATGCAAGTCGAACGATCCTTCGGGGTCGTGGCGTACGGCTCAGTAACACGTGGATAACCTAACCTTAGGACTGGGATAACCCTGGGAAACTGGGGATAATACCGGATATGTAGGGCTGCCTGGAATGGTTCCCTATTGAAATGTTCCGACGCCTAAGGATGGATCTGCGGCAGATTAGGTAGTTGGCGGGGTAAATGCCCACCAAGCCAGTAATCTGTACGGGTTGTGAGAGCAAGAGCCCGGAGATGGAACCTGAGACAAGGTTCCAGGCCCYACGGGGCGCAGCAGGCGCGAAACCTCCGCAATGCACGAAAGTGCGACGGGGGAAACCCAAGTGCCACTCTTAACGGGGTGGCTTTTCTTAAGTGTAAAAAGCTTTTGGAATAAGAGCTGGGCAAGACCGGTGCCAGCCGCCGCGGTAACACCGGCAGCTCAAGTGGTGGCCGTTTTTATTGGGCCTAAAGCGTTCGTAGCCGGCTTGATAAGTCTCTGGTGAAATCTCACGGCTTAACCGTGAGAATTGCTGGAGATACTATTAGGCTTGAGGCCGGGAGAGGTTAGCGGTACTCCCGGGGTAGGGGTGAAATCCTATAATCCCGGGAGGACCACCTGTGGCGAAGGCGGCTAACTGGAACGGACCTGACGGTGAGTAACGAAAGCCAGGGGCGCGAACCGGATTAGATACCCGGGTAGTCCTGGCCGTAAACGATGTGGACTTGGTGTTGGGATGGCTCCGAGCTGCCCCAGTGCCGAAGGGAAGCTGTTAAGTCCACCGCCTGGGAAGTACGGTCGCAAGACTGAAACTTAAAGGAATTGGCGGGGGAGCACCACAACGCGTGGAGCCTGCGGTTTAATTGGATTCAACGCCGGACATCTCACCAGGGGCGACAGCAGAATGATAGCCAGGTTGATGACCTTGCTTGACAAGCTGAGAGGAGGTGCATGGCCGCCGTCAGCTCGTACCGTGAGGCGTCCTGTTAAGTCAGGCAACGAGCGAGACCCACGCCCTTAGTTACCAGCGGATCCTTCGGGATGCCGGGCACACTAAGGGGACCGCCAGTGATAAACTGGAGGAAGGAGTGGACGACGGTAGGTCCGTATGCCCCGAATCCCCTGGGCTACACGCGGGCTACAATGGTTAGGACAATGGGTTCCGACACTGAAAGGTGGAGGTAATCTCCTAAACCTGGCCTTAGTTCGGATTGAGGGCTGTAACTCGCCCTCATGAAGCTGGAATGCGTAGTAATCGCGTGTCATAACCGCGCGGTGAATACGTCCCTGCTCCTTGCACACACCGCCCGTCACGCCACCCAAAAAGGGTTTGGATGAGGCCATAGTCTTTGGTTATGGTCGAATCTAGGTTCTTTG
```
{% endcode %}
{% endtab %}

{% tab title="Bacterial isolate B" %}
{% code overflow="wrap" %}
```
>Bacterial isolate B
TGDHTGATCCTGCCAGAGGTTACTGCTATCGGTGTTCGCCTAAGCCATGCGAGTCATATGTTCTTCGTGAACATGGCGTACTGCTCAGTAACACGTGGAKAACCTGCCCTTGGGTCTGGCATAACCCCGGGAAACTGGGGATAATTCCGGATAACGCATATATGCTGGAATGCTTTATGYGTAAAATGGATTCGTCCGCCCAAGNATGGGTCTGCGGCCTATCAGGTAGTAGTGGGTGTAATGTACCTACTAGCCTACAACGGGTACGGGTTGTGAGAGCAAGAGCCCGGAGATGGATTCTGAGACATGAATCCAGGCCCTACGGGNNGCAGCAGGCGCGAAAACTTTACAATGCGGGAAACCGTGATAAGGGGACACCGAGTGCTAGCATCATATGCTGGCTGTCCAGGTGTGTAAACTACACCTGTTAGCAAGGGCNGGGCAAGACCNGTGCCANNNGCCGCGGTAACACCGGNTGCCCGAGTGGTGATCGTGATTATTGGGTCTAAAGGGTCNGTAGCCGGTTTGGTCAGTCCTCCGGGAAATCTGATAGCTCAACTATTAGGCTTTCGGGGGATACTGCCAGACTTGGAACCGGGAGAGGTAAGAGGTACTACAGGGGTAGGAGTGAAATCTTGTAATCCCTGTGGGACCACCTGTGGCGAAGGCGTCTTACCAGAACGGGTTCGACGGTGAGGGACGAAAGCTGGGGGCACGAACCGGATTAGATACCCGGGTAGTCCCAGCCGTAAACGATGCTCGCTAGGTGTCAGGCATGGCGCGACCGTGTCTNGTGCCGCAGGGAAGCCGTGAAGCGAGCCACCTGGGAAGTACGGCCGCAAGGCTGAAACTTAAAGGAATTGGCGGGGGAGCACAACAACGGGTGGAGCCTGCGGTTTAATTGNNNNNAACGCCGGACAACTCACCGGGGACGACAGCAATATGTAGGTCAGGCTGAAGACCTTACCTGAATCGCTGAGAGGAGGTGCATGGCCGTCGCCNGTTCGTACTGTGAAGCATCCTGTTAAGTCAGGCAACGAGCGAGACCCGTGCCCACTGTTACCAGCATATCCTCCGGGATGATGGGTACTCTGTGGGNACCGCCGGTGTTAAATCGGAGGAAGGTGCGGGCCACGGTAGGTCAGTATGCCCCGAATTTCCCGGGCTACACGCGGGCTACAATGAATGGGACAATGGGTCCCTCCCCTGAAAAGGGCTGGTAATCTCACAAACCCATCCTTAGTTCGGATCGAGGGCTGTAACTCGCCCTCGTGAAGCTGGAATCCGTAGTAATCGCGTTTCAATATAGCGCGGTGAATACGTCCCTGCTCCTTGCACACACCGCCCGTCAAACCACCCGAGTGAGGTATGGGTGAGGGCACGAACATCGTGTCGTGTTCGAACCTGTGCTTTGCAAGGGGGGTTAAGNNGTAACAAGNNNNNNNNNNNNGAAT
```
{% endcode %}
{% endtab %}

{% tab title="Bacterial isolate C" %}
{% code overflow="wrap" %}
```
>Bacterial isolate C
AGTAHVACGTGGATAACCTGCCCTTGGGTCTGGCATAACCCCGGGAAACTGGGGATAATTCCGGATAACGCATATATGCTGGAATGCTTTATGCGTAAAATGGATTCGTCCGCCCAAGGATGGGTCTGCGGCCTATCAGGTAGTAGTGGGTGTAATGTACCTACTAGCCTACAACGGGTACGGGTTGTGAGAGCAAGAGCCCGGAGATGGATTCTGAGACATGAATCCAGGCCCTACGGGGCGCAGCAGGCGCGAAAACTTTACAATGCGGGAAACCGTGATAAGGGGACACCGAGTGCTAGCATCATATGCTGGCTGTCCAGGTGTGTAAACTACACCTGTTAGCAAGGGCCGGGCAAGACCGGTGCCAGCCGCCGCGGTAACACCGGCGGCCCGAGTGGTGATCGTGATTATTGGGTCTAAAGGGTCCGTAGCCGGTTTGGTCAGTCCTCCGGGAAATCTGATAGCTCAACTATTAGGCTTTCGGGGGATACTGCCAGACTTGGAACCGGGAGAGGTAAGAGGTACTACAGGGGTAGGAGTGAAATCTTGTAATCCCTGTGGGACCACCTGTGGCGAAGGCGTCTTACCAGAACGGGTTCGACGGTGAGGGACGAAAGCTGGGGGCACGAACCGGATTAGATACCCGGGTAGTCCCAGCCGTAAACGATGCTCGCTAGGTGTCAGGCATGGCGCGACCGTGTCTGGTGCCGCAGGGAAGCCGTGAAGCGAGCCACCTGGGAAGTACGGCCGCAAGGCTGAAACTTAAAGGAATTGGCGGGGGAGCACAACAACGGGTGGAGCCTGCGGTTTAATTGGACTCAACGCCGGACAACTCACCGGGGACGACAGCAATATGTAGGTCAGGCTGAAGACCTTACCTGAATCGCTGAGAGGAGGTGCATGGCCGTCGCCAGTTCGTACTGTGAAGCATCCTGTTAAGTCAGGCAACGAGCGAGACCCGTGCCCACTGTTACCAGCATATCCTCCGGGATGATGGGTACTCTGTGGGGACCGCCGGTGTTAAATCGGAGGAAGGTGCGGGCCACGGTAGGTCAGTATGCCCCGAATTTCCCGGGCTACACGCGGGCTACAATGAATGGGACAATGGGTCCCTCCCCTGAAAAGGGCTGGTAATCTCACAAACCCATCCTTAGTTCGGATCGAGGGCTGTAAWTCGCCCTCGTGAAGCTGGAATCCGTAGTAATCGCGTTTCAATATAGCGCGGTGAATACGTCCCTGCT
```
{% endcode %}
{% endtab %}

{% tab title="Bacterial isolate D" %}
{% code overflow="wrap" %}
```
>Bacterial isolate D
AGCAGGGACGTATTCACCGCGCTATATTGAAACGCGATTACTACGGATTCCAGCTTCACGAGGGCGANTTACAGCCCTCGATCCGAACTAAGGATGGGTTTGTGAGATTACCAGCCCTTTTCAGGGGAGGGACCCATTGTCCCATTCATTGTAGCCCGCGTGTAGCCCGGGAAATTCGGGGCATACTGACCTACCGTGGCCCGCACCTTCCTCCGATTTAACACCGGCGGTCCCCACAGAGTACCCATCATCCCGGAGGATATGCTGGTAACAGTGGGCACGGGTCTCGCTCGTTGCCTGACTTAACAGGATGCTTCACAGTACGAACTGGCGACGGCCATGCACCTCCTCTCAGCGATTCAGGTAAGGTCTTCAGCCTGACCTACATATTGCTGTCGTCCCCGGTGAGTTGTCCGGCGTTGAGTCCAATTAAACCGCAGGCTCCACCCGTTGTTGTGCTCCCCCGCCAATTCCTTTAAGTTTCAGCCTTGCGGCCGTACTTCCCAGGTGGCTCGCTTCACGGCTTCCCTGCGGCACCAGACACGGTCGCGCCATGCCTGACACCTAGCGAGCATCGTTTACGGCTGGGACTACCCGGGTATCTAATCCGGTTCGTGCCCCCAGCTTTCGTCCCTCACCGTCGAACCCGTTCTGGTAAGACGCCTTCGCCACAGGTGGTCCCACAGGGATTACAAGATTTCACTCCTACCCCTGTAGTACCTCTTACCTCTCCCGGTTCCAAGTCTGGCAGTATCCCCCGAAAGCCTAATAGTTGAGCTATCAGATTTCCCGGAGGACTGACCAAACCGGCTACGGACCCTTTAGACCCAATAATCACGATCACCACTCGGGCCGCCGGTGTTACCGCGGCGGCTGGCACCGGTCTTGCCCGGCCCTTGCTAACAGGTGTAGTTTACACACCTGGACAGCCAGCATATGATGCTAGCACTCGGTGTCCCCTTATCACGGTTTCCCGCATTGTAAAGTTTTCGCGCCTGCTGCGCCCCGTAGGGCCTGGATTCATGTCTCAGAATCCATCTCCGGGCTCTTGCTCTCACAACCCGTACCCGTTGTAGGCTAGTAGGTACATTACACCCACTACTACCTGATAGGCCGCAGACCCATCCTTGGGCGGACGAATCCATTTTACGCATAAAGCATTCCAGCATATATGCGTTATCCGGAATTATCCCCAGTTTCCCGGGGTTATGCCAGACCCAAGGGCAGGTTATCCACGTNNTACT
```
{% endcode %}
{% endtab %}
{% endtabs %}

### Questions

* Look at the [sequencing results](#user-content-fn-1)[^1]. We see nucleotides that are not `A`, `T`, `C`, or `G`. What are they and what do they represent?
* Make sure to review the benchtop methods. Walk yourself through the protocols. What sources of technical errors might arise that could affect your sequencing results?
* Use NCBI BLASTN ([https://blast.ncbi.nlm.nih.gov/Blast.cgi](https://blast.ncbi.nlm.nih.gov/Blast.cgi)) to identify your samples. Compare the results. Make sure to consider how confident these results are.
* What is something common about each of the bacteria that were isolated?
* Let's suppose that we want to focus on specific strains. What do you notice from the BLAST results for strain-level identification?
* Write a brief description of an additional three "real world" examples of species identification using DNA barcoding.&#x20;

<details>

<summary>Further Questions</summary>

1. Some of the bacteria may be closely related. Is there a technique to identify how closely related they might be?
2. Are there any limitations to the fact that we are culturing and plating to identify these bacteria?
3. Double-peaks and degenerate nucleotides are sometimes evidence of heterozygosity. Is this something you would expect in bacteria? Why or why not?
4. We are currently sequencing bacteria to identify them one at a time. Is there a way that we might be able to sequence all the bacteria in the bioreactor to get a better idea of its composition?

</details>

[^1]: Note that they are in FASTA-format.&#x20;
